### PR TITLE
fix(vue): Update data setup guide for Vue, remove tsx 

### LINKED ISSUE #7622: Vue documentation importing React?
- Amplify Docs Gen2 - Set up Amplify Data - for Vue
- The supplied code references react multiple times e.g 'src/TodoList.tsx' top line is 'import { useState, useEffect } from "react"; and 'src/App.tsx' second line 'import { useState, useEffect } from "react";'

URL page where content issue is: https://docs.amplify.aws/vue/build-a-backend/data/set-up-data/

### PR DESCRIPTION
**Pull request by ErikCH**
#### Description of changes:
The data setup guide for Vue is not correct. It shows React code instead of Vue code.

#### Related GitHub issue #, if available:

Fixes #7622 

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [x] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx [hunk 1]
```diff
@@ -139,6 +142,20 @@ Amplify.configure(outputs);
 
 </InlineFilter>
 
+<InlineFilter filters={["vue"]}>
+
+In your app's entry point, typically **main.ts** for Vue apps created using Vite, make the following edits:
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx:147]: the previous example should show `main.ts`. Vite does not create a tsx file for non-vue projects

File: src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx [hunk 2]
```diff
@@ -380,6 +402,40 @@ export default function TodoList() {
 }
 ```
 
+</InlineFilter>
+
+
+<InlineFilter filters={["vue"]}>
+
+```html title="src/TodoList.vue"
+
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx:411]:
```suggestion
```

File: src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx [hunk 3]
```diff
@@ -380,6 +402,40 @@ export default function TodoList() {
...
+  })
+}
+</script>
+
+<template>
+  <div>
+    <button @click="createTodo">Add new todo</button>
+  </div>
+</template>
+
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx:431]:
```suggestion
```

File: src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx [hunk 4]
```diff
@@ -623,6 +679,56 @@ export default function TodoList() {
 }
 ```
 
+</InlineFilter>
+
+<InlineFilter filters={["vue"]}>
+
+```html title="src/TodoList.vue"
+
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx:687]:
```suggestion
```

File: src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx [hunk 5]
```diff
@@ -623,6 +679,56 @@ export default function TodoList() {
...
+    <ul>
+     <li 
+       v-for="todo in todos" 
+       :key="todo.id">
+       {{ todo.content }}
+     </li>
+    </ul>
+  </div>
+</template>
+
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx:729]:
```suggestion
```

File: src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx [hunk 6]
```diff
@@ -623,6 +679,56 @@ export default function TodoList() {
...
+<script setup lang="ts">
+import { onMounted, ref } from 'vue';
+import type { Schema } from '../../amplify/data/resource'
+import { generateClient } from 'aws-amplify/data'
+
+const client = generateClient<Schema>()
+
+// create a reactive reference to the array of todos
+const todos = ref<Array<Schema['Todo']["type"]>>([]);
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx:696]: quotes should be consistent -- single or double

File: src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx [hunk 7]
```diff
@@ -880,6 +986,63 @@ export default function TodoList() {
 }
 ```
 
+</InlineFilter>
+
+<InlineFilter filters={["vue"]}>
+```html title="src/TodoList.vue"
+
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx:993]:
```suggestion
```

File: src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx [hunk 8]
```diff
@@ -880,6 +986,63 @@ export default function TodoList() {
...
+    <ul>
+     <li 
+       v-for="todo in todos" 
+       :key="todo.id">
+       {{ todo.content }}
+     </li>
+    </ul>
+  </div>
+</template>
+
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx:1039]:
```suggestion
```

File: src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx (hunk 1 above)
Comment (ErikCH) [reply src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx:147]: 👍

File: src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx (hunk 2 above)
Comment (ErikCH) [reply src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx:415]: 👍

File: src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx (hunk 4 above)
Comment (ErikCH) [reply src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx:691]: 👍

File: src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx (hunk 6 above)
Comment (ErikCH) [reply src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx:700]: 👍

File: src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx (hunk 7 above)
Comment (ErikCH) [reply src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx:997]: 👍

File: src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx (hunk 8 above)
Comment (ErikCH) [reply src/pages/[platform]/build-a-backend/data/set-up-data/index.mdx:1043]: 👍

**Review by josefaidt — APPROVED** [review]

**Review by chrisbonifacio — APPROVED** [review]

**Review by reesscot — APPROVED** [review]

**Pull request merged**